Build failure: intel-llvm

### Nixpkgs version

- Unstable (26.11)

### Steps to reproduce

build it on master (723991338737)

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/336750940

### Relevant log output

```console
[4234/5673] Generating ../../lib/fallback-imf-fp64-host.new.o
FAILED: [code=1] lib/fallback-imf-fp64-host.new.o /build/source/build/lib/fallback-imf-fp64-host.new.o 
cd /build/source/build/tools/libdevice && /nix/store/461bw93pijjc8109g9saqd9lmn9hnfd4-ccWrapperStub-wrapper-/bin/clang++ -Wno-sycl-strict -Wno-undefined-internal -sycl-std=2020 --target=x86_64-pc-linux-gnu -c --target=x86_64-pc-linux-gnu -D__LIBDEVICE_HOST_IMPL__ -fPIC -fcf-protection=full --offload-new-driver -foffload-lto=thin -I /build/source/libdevice/imf /build/source/build/./lib/libdevice/imf_fp64_fallback.cpp -o /build/source/build/./lib/fallback-imf-fp64-host.new.o
Warning: supplying the --target x86_64-pc-linux-gnu != x86_64-unknown-linux-gnu argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
In file included from /build/source/build/./lib/libdevice/imf_fp64_fallback.cpp:14:
In file included from /build/source/libdevice/imf/../device_imf.hpp:13:
In file included from /build/source/libdevice/imf/../imf_bf16.hpp:13:
/build/source/libdevice/imf/../imf_impl_utils.hpp:12:10: fatal error: 'cstddef' file not found
   12 | #include <cstddef>
      |          ^~~~~~~~~
1 error generated.
[4235/5673] Generating obj.libclc.dir/native_cpu/opencl/lib/generic/shared/vload.cl.bc
[4236/5673] Generating ../../lib/nativecpu_utils.bc
FAILED: [code=1] lib/nativecpu_utils.bc /build/source/build/lib/nativecpu_utils.bc 
cd /build/source/build/tools/libdevice && /nix/store/461bw93pijjc8109g9saqd9lmn9hnfd4-ccWrapperStub-wrapper-/bin/clang++ -Wno-sycl-strict -Wno-undefined-internal -sycl-std=2020 --target=x86_64-pc-linux-gnu -fsycl-device-only -fsycl-device-obj=llvmir -fsycl-targets=native_cpu -fno-sycl-libspirv -Wno-unsafe-libspirv-not-linked -I /build/source/build/include -I /build/source/unified-runtime/source/adapters/native_cpu /build/source/libdevice/nativecpu_utils.cpp -o /build/source/build/./lib/nativecpu_utils.bc
Warning: supplying the --target x86_64-pc-linux-gnu != x86_64-unknown-linux-gnu argument to a nix-wrapped compiler may not work correctly - cc-wrapper is currently not designed with multi-target compilers in mind. You may want to use an un-wrapped compiler instead.
clang-22: warning: argument unused during compilation: '-fstack-clash-protection' [-Wunused-command-line-argument]
clang-22: warning: argument unused during compilation: '-mno-omit-leaf-frame-pointer' [-Wunused-command-line-argument]
clang-22: warning: argument unused during compilation: '-Wl,-dynamic-linker=/nix/store/ias8xacs1h3jy7xgwi2awvim61k2ji6c-glibc-2.42-67/lib/ld-linux-x86-64.so.2' [-Wunused-command-line-argument]
clang-22: warning: argument unused during compilation: '-Wl,-rpath' [-Wunused-command-line-argument]
clang-22: warning: argument unused during compilation: '-Wl,/nix/store/c2pazp4p8h8xq2k6dw4rclr061z5zvzf-intel-llvm-unstable-2025-11-14-lib/lib' [-Wunused-command-line-argument]
/build/source/libdevice/nativecpu_utils.cpp:18:10: fatal error: 'cstdint' file not found
   18 | #include <cstdint>
      |          ^~~~~~~~~
```

### Additional context

_No response_

### System metadata

x86_64-linux

### Notify maintainers

@kilyanni

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.
- [x] I assert that I have read the [automation/AI policy](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#automationai-policy) and that this issue report complies with it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/in/intel-llvm/unwrapped.nix
+++ b/pkgs/by-name/in/intel-llvm/unwrapped.nix
@@ -57,7 +57,11 @@ let
         cp $out/bin/clang-${llvmMajorVersion} $out/bin/clang
         cp $out/bin/clang-${llvmMajorVersion} $out/bin/clang++
       '';
-      passthru.isClang = true;
+      passthru = {
+        isClang = true;
+        langC = true;
+        langCC = true;
+      };
     }
   );
 
@@ -296,6 +300,8 @@ stdenv.mkDerivation (finalAttrs: {
 
   passthru = {
     isClang = true;
+    langC = true;
+    langCC = true;
 
     inherit unified-runtime;
 

